File: src/turtle_control/scripts/turtle_pose.py
import rospy

from turtlesim.msg import Pose


def get_pose(pos):
    rospy.loginfo(f"x= %f y= %f theta= %f", pos.x, pos.y, pos.theta)

if __name__ == "__main__":
    rospy.init_node("turtle_pose_py")
    rospy.Subscriber("/turtle1/pose", Pose, get_pose, queue_size=10)

    rospy.spin()

File: src/turtle_control/src/turtle_pose.cpp
#include "ros/ros.h"
#include "turtlesim/Pose.h"

void get_pose(const turtlesim::Pose::ConstPtr& pos)
{
    ROS_INFO("x = %f", pos->x);
    ROS_INFO("y = %f", pos->y);
    ROS_INFO("theta = %f", pos->theta);
}

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "turtle_pose");
    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe("/turtle1/pose", 10, get_pose);

    ros::spin();
    return 0;
}
